storage/plugins/mobile-api/src/Controllers/PushController.php

```
@@ -0,0 +1,422 @@
+
+            return ResponseEnvelope::success(
+                $response,
+                ['id' => $subId, 'provider' => $provider],
+                ['message' => __('Dispositivo registrato per le notifiche push.')],
+                201
+            );
+        } catch (\Throwable $e) {
+            SecureLogger::error('[MobileApi] push subscribe failed: ' . $e->getMessage());
+            return ResponseEnvelope::error($response, 'internal_error', __('Operazione non disponibile.'), 500);
+        }
+    }
```

PHPMD (2.15.0)

[warning] The method subscribe() has a Cyclomatic Complexity of 19. The configured cyclomatic complexity threshold is 10. (undefined)

(CyclomaticComplexity)